http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -696,13 +737,30 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       )
     } yield ()
 
-  def handleBatchNodepoolCreateMessage(msg: BatchNodepoolCreateMessage)(
+  private[monitor] def handleBatchNodepoolCreateMessage(msg: BatchNodepoolCreateMessage)(
     implicit ev: ApplicativeAsk[F, AppContext]
   ): F[Unit] =
     for {
       ctx <- ev.ask
-      initialGoogleCall = gkeInterp.createCluster(msg.clusterId, msg.nodepools, true)
-      clusterResult <- initialGoogleCall.adaptError {
+
+      // Unlike handleCreateAppMessage, here we assume that the cluster does not exist so we always create it.
```

Comment:
Does it make sense to check if the cluster exists before going through with the action?

Reply:
I think front Leo checks this, it throws 409 on `batchNodepoolCreate` if a cluster already exists for the project.

However it might be a good idea to add a check here for safety as well..

Reply:
Actually I don't think we need an additional check -- the `GKEInterpreter` call will fail appropriately if the cluster already exists.